### DISCUSSION TITLE
Admin console collection drop down leaking other accounts endpoints

### DISCUSSION THREAD

**Discussion by stevengibbsonline** [swellstores/community#256] [Help] [answered]
**Issue Description:**
When accessing Swell's Developer Console via the Admin Web Interface, endpoints that do not belong to our account are displayed. This raises concerns about data privacy and security as it exposes potentially sensitive information and poses a risk of unauthorized access to our account.

**Steps to Reproduce:**

1. Log in to Swell's Admin Web Interface using valid credentials.
2. Navigate to the Developer Console section.
3. Observe that endpoints and API resources that do not correspond to our account are visible.

**Expected Behavior:**
The Developer Console collection dropdown should only display endpoints and resources that are relevant and accessible to our specific account. Unauthorized endpoints and resources should not be visible.

**Actual Behavior:**
The Drop Down shows endpoints that do not belong to our account, potentially exposing sensitive information about unauthorized resources.

**Additional Information:**

This issue poses a serious security risk as it allows information gathering on endpoints and resources that should be restricted.
It compromises data privacy and confidentiality.
This issue is persistent and occurs consistently whenever accessing the Console.
We have confirmed that our account permissions and settings are properly configured, and the issue persists regardless.

**Proposed Solution:**
It is recommended that Swell's development team investigates and resolves this issue promptly to ensure 
![346162489_535038448634620_7223260963191660984_n](https://github.com/swellstores/community/assets/41846158/38cdb72c-2aba-4f5d-a64e-31fc260127f2)
![346168852_267756335727667_1423507261144256476_n](https://github.com/swellstores/community/assets/41846158/0270ba47-8357-4868-87a9-95cb4353ad9b)
![348361550_754169486450983_1097398280810895471_n](https://github.com/swellstores/community/assets/41846158/5e59a3b1-2ceb-4895-821b-31c287070d52)
the privacy and security of our account and data. The API Explorer should be restricted to displaying only endpoints and resources that are associated with the authenticated account.

**Relevant Context:**

Web browser and version used: Chrome Version 113.0.5672.127 (Official Build) (64-bit)
Operating system used: Windows 10, Windows 11

Please let me know if there is any additional information required for the GitHub issue.

**Comment by steven-k1** [comment]
@stevengibbsonline Would you be able to confirm the ID of the account so that we can look into this?

**Comment by steven-k1** [comment, marked as answer]
@stevengibbsonline We looked into the raised concern and confirmed an issue, but the issue is only leaking model names and not data to be clear. We'll have this worked on and get back to you as updates become available.

**Reply by stevengibbsonline** [reply]
Sorry for the delay in responding - it has been a busy week. 

That's great to hear. Thank you @StevenSwell

**Reply by steven-k1** [reply]
@stevengibbsonline A fix for this issue has been deployed to production.

Please let us know if you continue to experience any related issues.
